### ISSUE TITLE
Issues with finfo

### ISSUE THREAD

**Issue by Mark-H** [darkstardigital/phpThumbsUp#16]
Apparently PhpThumbsUp requires the fileinfo extension which is not installed on every server. This may very well be related to #11 as in the past I had to enable specific extensions on wamp. 

The specific error I'm seeing (on a linux server) is the following:

Fatal error: Call to undefined function finfo_open() in /home/user/core/components/phpthumbsup/model/phpthumbsup/phpthumbsup.class.php on line 417

It would be best to check if the function exists, and fall back to other functions instead. This likely affects all instances of finfo_*.

**Comment by Mark-H** [comment]
This code to get the mime type is breaking it:

```
$finfo = finfo_open(FILEINFO_MIME_TYPE);
$mime = finfo_file($finfo, $file);
finfo_close($finfo);
```

To provide some ideas, this is how I get the mime type within MoreGallery taking advantage of the exif extension if available, or using getimagesize otherwise.

```
// Get the mime type
if (function_exists('exif_imagetype')) {
    $imageType = exif_imagetype($array['mgr_thumb_path']);
} else {
    $imageType = getimagesize($array['mgr_thumb_path']);
    $imageType = $imageType[2];
}
$mime = image_type_to_mime_type($imageType);
```

**Comment by sdwespi** [comment]
Hey Mark,

When we created phpThumbsUp we decided to target PHP 5.3+ (finfo is included in PHP 5.3.0+) and that's been documented in our minimum requirements. We may explore expanding our support to cover PHP 5.2.X in the future, but right now our focus is on enhancing and extending our existing feature set.

Thanks for the suggestion!

-Scott

**Comment by Mark-H** [comment]
Hi Scott!

Both my server and localhost ate running php 5.4.4 and do not have finfo
compiled in. I don't think finfo is included by default but that can differ
per distribution I guess.
Op 22 aug. 2013 21:39 schreef "Scott Wespi" notifications@github.com het
volgende:

**Comment by patrickmckay** [comment]
Are you running on a Linux distro or Windows? According to the PHP documentation the finfo extensions are enabled by default on versions 5.3+ ( http://www.php.net/manual/en/fileinfo.installation.php )

However, it does say "Windows users must include the bundled php_fileinfo.dll DLL file in php.ini to enable this extension".

We will be releasing 1.1.2 in the near future and will add a fallback to this release in case finfo is not enabled.

**Comment by Mark-H** [comment]
Linux and mamp on mac. On the live (linux) server, apparently fileinfo is disabled from the configure command. I don't know why (could ask the host) but all in all I think a fallback would be great ;)

**Comment by davidpede** [comment]
+1 as I ran into this issue as well

**Comment by sdwespi** [comment]
We've actually fixed this issue in 1.1.2, but we haven't posted a transport package to MODX yet. We were hoping to get a few more changes in and post that release to MODX, but if this issue is holding you up you can always download 1.1.2 from github. The next release should be ready soon.

Thanks for your patience!

**Comment by abelafonte** [comment]
Just downloaded from Github and I can confirm that this is fixed in 1.1.2.

**Comment by micheldejoode** [comment]
Hi abelafonte, i've downloaded 1.1.2 too from github, but when I place the zip file in core/packages and search for local packages, Modx won't find it. When I change the filename to .transport.zip, Modx finds it, but says there is no manifest file at install. I also unzipped the file and place the folder in core/packages, but it gives the same problems. Any ideas how to do this?
